[TextFields] Refactor snapshot tests into abstract class.

romoore opened the pull request:
To avoid having to duplicate test methods across different input controllers
(e.g., outlined or full-width) and text field types (e.g., single- or
multi-line), we can move the actual test logic into an abstract class that
only needs to know about the text properties to be set on the controller or
text input.

Part of #5762

material-automation commented on components/TextFields/tests/snapshot/MDCTextFieldOutlinedControllerBaselineCharacterCountSnapshotTests.m:
```diff
@@ -41,6 +41,9 @@ - (void)setUp {
   self.textField.clearButtonMode = UITextFieldViewModeAlways;
+
+  self.textFieldController =
+      [[MDCTextInputControllerOutlined alloc] initWithTextInput:self.textField];;
```
clang-format suggested the following change:


```
-      [[MDCTextInputControllerOutlined alloc] initWithTextInput:self.textField];;
+      [[MDCTextInputControllerOutlined alloc] initWithTextInput:self.textField];
+  ;
```

<!-- identifier: clang-format -->
<!-- File: components/TextFields/tests/snapshot/MDCTextFieldOutlinedControllerBaselineCharacterCountSnapshotTests.m -->

material-automation commented on components/TextFields/tests/snapshot/MDCTextFieldOutlinedControllerBaselineSnapshotTests.m:
```diff
@@ -12,15 +12,15 @@
 
 @interface MDCTextFieldOutlinedControllerBaselineSnapshotTests
-    : MDCTextFieldOutlinedControllerSnapshotTests
+    : MDCAbstractTextFieldSnapshotTests
```
clang-format suggested the following change:


```
-@interface MDCTextFieldOutlinedControllerBaselineSnapshotTests
-    : MDCAbstractTextFieldSnapshotTests
+@interface MDCTextFieldOutlinedControllerBaselineSnapshotTests : MDCAbstractTextFieldSnapshotTests
```

<!-- identifier: clang-format -->
<!-- File: components/TextFields/tests/snapshot/MDCTextFieldOutlinedControllerBaselineSnapshotTests.m -->

material-automation commented:
clang-format recommended changes.

Consider installing [git-clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) and running the following command:

```bash
git clang-format $(git merge-base origin/develop HEAD)
```

<!-- identifier: clang-format -->

material-automation commented on components/TextFields/tests/snapshot/MDCTextFieldFilledControllerSnapshotTests.m:
```diff
@@ -12,12 +12,11 @@
-#import "SnapshotFakeMDCTextField.h"
 
-@interface MDCTextFieldFilledControllerSnapshotTests : MDCTextFieldSnapshotTestCase
+@interface MDCTextFieldFilledControllerSnapshotTests :  MDCAbstractTextFieldSnapshotTests
```
clang-format suggested the following change:


```
-@interface MDCTextFieldFilledControllerSnapshotTests :  MDCAbstractTextFieldSnapshotTests
+@interface MDCTextFieldFilledControllerSnapshotTests : MDCAbstractTextFieldSnapshotTests
```

<!-- identifier: clang-format -->
<!-- File: components/TextFields/tests/snapshot/MDCTextFieldFilledControllerSnapshotTests.m -->

Commit:
[TextFields] Refactor snapshot tests into abstract class.

To avoid having to duplicate test methods across different input controllers
(e.g., outlined or full-width) and text field types (e.g., single- or
multi-line), we can move the actual test logic into an abstract class that
only needs to know about the text properties to be set on the controller or
text input.

Part of #5762

Commit:
Merge branch 'develop' into issue-5762-e
